ledger/ledger-api-tests/infrastructure/src/main/scala/com/daml/ledger/api/testtool/infrastructure/Assertions.scala

```diff
@@ -69,41 +69,13 @@ object Assertions {
     assertSameElements(actual, Seq.empty)
   }
 
-  case class ExtendedAsserts(
-      exceptionMessageSubstring: Option[String] = None,
-      checkDefiniteAnswerMetadata: Boolean = false,
-      additionalErrorAssertions: Throwable => Unit = _ => (),
-  )
-
-  def assertGrpcErrorOneOf(
-      t: Throwable,
-      extendedChecks: Option[ExtendedAsserts],
-      errors: ErrorCode*
-  ): Unit = {
-    errors.map(errorCode => (errorCode, Try(assertGrpcError(t, errorCode, None)))).collectFirst {
-      case (errorCode, Success(_)) => errorCode
-    } match {
-      case Some(matchedErrorCode) =>
-        // In order to catch errors related to the "extended checks" rerun the check on the matching error code.
-        // This results in a more specific error message if one of the error codes matches, but the other checks fail.
-        extendedChecks.foreach {
-          case ExtendedAsserts(
-                exceptionMessageSubstring,
-                checkDefiniteAnswerMetadata,
-                additionalErrorAssertions,
-              ) =>
-            assertGrpcError(
-              t,
-              matchedErrorCode,
-              exceptionMessageSubstring,
-              checkDefiniteAnswerMetadata,
-              additionalErrorAssertions,
-            )
-        }
-
-      case None =>
-        fail(s"gRPC failure did not contain one of the expected error codes $errors.", t)
+  def assertGrpcErrorOneOf(t: Throwable, errors: ErrorCode*): Unit = {
+    val hasErrorCode = errors.map(errorCode => Try(assertGrpcError(t, errorCode, None))).exists {
+      case Success(_) => true
+      case _ => false
     }
+    if (hasErrorCode) ()
+    else fail(s"gRPC failure did not contain one of the expected error codes $errors.", t)
```

Comment:
Can we simplify by using a negation here?
```suggestion
    if (!hasErrorCode) fail(s"gRPC failure did not contain one of the expected error codes $errors.", t)
```